kernel/term_typing.ml

```diff
@@ -176,8 +176,12 @@ let infer_declaration env (dcl : constant_entry) =
            Vars.subst_univs_level_constr usubst tj.utj_val
       in
       let def = Vars.subst_univs_level_constr usubst j.uj_val in
-      let def = Def (Mod_subst.from_val def) in
-        feedback_completion_typecheck feedback_id;
+      let def =
+        if c.const_entry_opaque then
+          Obj.magic @@ OpaqueDef ()
```

Comment:
Wut?

Reply:
please, read my comments ;-)

Reply:
Sorry, race condition. I was reading the code while you were writing the comment :/